spikeinterface/widgets/template_similarity.py

```diff
@@ -41,6 +44,9 @@ def __init__(self, waveform_extractor: WaveformExtractor,
             unit_indices = sorting.ids_to_indices(unit_ids)
             similarity = similarity[unit_indices][:, unit_indices]
 
+        if not display_diagonal_values:
+            np.fill_diagonal(similarity, 0)
```

Comment:
I think I really prefer this solution rather than hacking the computation function itself .